src/v/cluster/archival/async_data_uploader.cc

```diff
@@ -343,7 +343,12 @@ segment_upload::compute_upload_parameters(
           .size_bytes = sz->on_disk_size,
           .is_compacted = _part->log()->is_compacted(
             range_base, sz->last_offset),
+          .eligible_for_compacted_reupload
+          = _part->log()->eligible_for_compacted_reupload(
+            range_base, sz->last_offset),
           .offsets = inclusive_offset_range(range_base, sz->last_offset),
+          .base_timestamp = sz->first_timestamp,
+          .max_timestamp = sz->last_timestamp,
```

Comment:
For the folks (like me) who don't have this region of code cached, mind adding to the commit message what's the context this code is currently used in, and how do these new changes fit in?